packages/react-native-views/src/components/Thumbnail/index.js

```diff
@@ -0,0 +1,146 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { propTypes, sightMasks, utils } from '@monkvision/react-native';
+import { Image, View, StyleSheet } from 'react-native';
+import { ActivityIndicator, withTheme, IconButton } from 'react-native-paper';
+
+import useToggle from '../../hooks/useToggle';
+import ComplianceModal from './ComplianceModal';
+
+const { spacing } = utils.styles;
+const styles = StyleSheet.create({
+  sightCard: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    marginVertical: spacing(1),
+    borderRadius: 22,
+    position: 'relative',
+    borderWidth: 1.2,
+    margin: 4,
+    width: 100,
+    height: 100,
+  },
+  picture: {
+    width: '100%',
+    height: '100%',
+    borderRadius: 22,
+    marginVertical: 4,
+    position: 'absolute',
+    zIndex: 0,
+    opacity: 0.2,
+  },
+  loading: {
+    position: 'absolute',
+    alignSelf: 'center',
+  },
+  reloadButtonLayout: {
+    position: 'absolute',
+    display: 'flex',
+    flexDirection: 'row',
+    zIndex: 11,
+    alignSelf: 'center',
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+  overlayButton: {
+    backgroundColor: '#FFF',
+    margin: spacing(1),
+  },
+});
+
+const SightMask = ({ id, ...props }) => (sightMasks?.[id] ? sightMasks[id](props) : null);
+
+const colorsVariant = (colors) => ({
+  rejected: colors.error,
+  pending: colors.primary,
+  fulfilled: colors.primary,
+});
+function Thumbnail({ theme, complianceStatus, image, isUploading, metadata, uploadStatus }) {
+  const { colors } = theme;
+  const { id } = metadata.sight;
+
+  const [complianceModalIsOpen, openComplianceModal, closeComplianceModal] = useToggle();
+  const complianceIssues = useMemo(
+    () => Object.keys(complianceStatus).filter(
+      (key) => !!complianceStatus[key],
+    ), [complianceStatus],
+  );
+  const isNotCompliant = complianceIssues?.length;
+
+  console.log({ complianceStatus, image, isUploading, uploadStatus });
+
+  const isFailed = uploadStatus === 'rejected';
+  const isPending = uploadStatus === 'pending';
+  return (
```

Comment:
Always separate the return statement with a blank line. (I hesitate to add an eslint rule for that).